docs/topics/introduction.md

```diff
@@ -0,0 +1,269 @@
+
+## Trace the invalid execution
+
+Besides showing invalid execution results, Lincheck can also provide an interleaving that leads to the error. This
+feature is accessible with the [model checking](testing-strategies.md#model-checking) testing strategy,
+which examines numerous executions with a bounded number of context switches.
+
+1. To switch the testing strategy, replace the `options` type from `StressOptions()` to `ModelCheckingOptions()`.
+The updated `CounterTest` class will look like this:
+
+   ```kotlin
+   import org.jetbrains.kotlinx.lincheck.annotations.*
+   import org.jetbrains.kotlinx.lincheck.check
+   import org.jetbrains.kotlinx.lincheck.strategy.managed.modelchecking.*
+   import org.jetbrains.kotlinx.lincheck.verifier.*
+   import org.junit.*
+   
+   class Counter {
+       @Volatile
+       private var value = 0
+   
+       fun inc(): Int = ++value
+       fun get() = value
+   }
+   
+   class BasicCounterTest {
+       private val c = Counter()
+   
+       @Operation
+       fun getAndInc() = c.getAndInc()
+   
+       @Operation
+       fun get() = c.get()
+   
+       @Test
+       fun modelCheckingTest() = ModelCheckingOptions().check(this::class)
+   }
+   ```
+
+2. Run the test again. You will get the execution trace that leads to incorrect results:
+
+   ```text
+   = Invalid execution results =
+   Parallel part:
+   | inc(): 1 | inc(): 1 |
+   = The following interleaving leads to the error =
+   Parallel part trace:
+   |                      | inc()                                                      |
+   |                      |   inc(): 1 at BasicCounterTest.inc(BasicCounterTest.kt:11) |
+   |                      |     value.READ: 0 at Counter.inc(Counter.kt:5)             |
+   |                      |     switch                                                 |
+   | inc(): 1             |                                                            |
+   |   thread is finished |                                                            |
+   |                      |     value.WRITE(1) at Counter.inc(Counter.kt:5)            |
+   |                      |     value.READ: 1 at Counter.inc(Counter.kt:5)             |
+   |                      |   result: 1                                                |
+   |                      |   thread is finished                                       |
+   ```
+
+   According to the trace, the following events have occurred:
+
+   * **T2**: The second thread starts the `inc()` operation, reading the current counter value (`value.READ: 0`) and pausing.
+   * **T1**: The first thread executes `inc()`, which returns `1`, and finishes.
+   * **T2**: The second thread resumes and increments the previously obtained counter value, incorrectly updating the
+   counter to `1`.
+
+> Get the full code [here](https://github.com/Kotlin/kotlinx-lincheck/blob/guide/src/jvm/test/org/jetbrains/kotlinx/lincheck/test/guide/BasicCounterTest.kt).
```

Comment:
It's better not to use the "here" word as a link (for the sake of SEO)
[Get the full code] will be better as a link